h2/test_h2_headers.py

```diff
@@ -106,3 +107,184 @@ def test_random_header(self):
         self.start_tempesta()
         self.start_all_clients()
         self.wait_while_busy(curl)
+
+class CurlTestBase(tester.TempestaTest):
+
+    clients = [
+        {
+            'id' : 'curl',
+            'type' : 'external',
+            'binary' : 'curl',
+            'cmd_args' : (
+                '-kf ' # Set non-null return code on 4xx-5xx responses.
+                'https://${tempesta_ip}/ '
+                )
+        },
+    ]
+
+    def test(self, served_from_cache=False):
+        curl = self.get_client('curl')
+
+        self.start_all_servers()
+        self.start_tempesta()
+
+        self.start_all_clients()
+        self.wait_while_busy(curl)
+        self.assertEqual(0, curl.returncode,
+                         msg=("Curl return code is not 0 (%d)." %
+                              (curl.returncode)))
+        curl.stop()
+
+        self.start_all_clients()
+        self.wait_while_busy(curl)
+        self.assertEqual(0, curl.returncode,
+                         msg=("Curl return code is not 0 (%d)." %
+                              (curl.returncode)))
+
+        nginx = self.get_server('nginx')
+        nginx.get_stats()
+        self.assertEqual(1 if served_from_cache else 2, nginx.requests)
+
+class AddBackendShortHeaders(CurlTestBase):
+    ''' The test checks the correctness of forwarding short headers with
+    duplication in mixed order: put header B between two headers A
+    '''
+
+    backends = [
+        {
+            'id' : 'nginx',
+            'type' : 'nginx',
+            'port' : '8000',
+            'status_uri' : 'http://${server_ip}:8000/nginx_status',
+            'config' : NGINX_CONFIG % """
+add_header x-extra-data1 "q";
+add_header x-extra-data2 "q";
+add_header x-extra-data1 "q";
+
+return 200;
+""",
+        }
+    ]
+
+    tempesta = {
+        'config' : TEMPESTA_CONFIG % "",
+    }
+
+    def test(self):
+        CurlTestBase.test(self)
```

Comment:
Seems like there is no need to override `test()` function from `CurlTestBase` class here.